### PR TITLE
paramdigger: Add naming changes to GUI and helpdocs

### PR DESCRIPTION
**Pull request by ArkaprabhaChakraborty**
Related to zaproxy/zaproxy#7140 and fixes zaproxy/zaproxy#7427
Signed-off-by: ArkaprabhaChakraborty <chakrabortyarkaprabha998@gmail.com>

### CODE REVIEW

**Comment by ArkaprabhaChakraborty** [comment]
@kingthorin @thc202

**Review by kingthorin — APPROVED** [review]
LGTM

**Comment by kingthorin** [comment]
CHANGELOG.md

**Comment by ArkaprabhaChakraborty** [comment]
Uhh @thc202 said its not really released yet so might not be necessary :). But ... :) ... I might have gotten it wrong tho :)

**Comment by kingthorin** [comment]
Okay, see what others say. I’m not overly picky.

**Comment by thc202** [comment]
What I was saying is that the add-on will probably have more changes that this one not worth mentioning (but the changelog should be updated always, at least to have "Maintenance changes."). But since there's an issue maybe mention it…

**Comment by kingthorin** [comment]
A few other tweaks I had ideas on:

```diff
$ git diff
diff --git a/addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/ExtensionParamMiner.java b/addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/ExtensionParamMiner.java
index e1db62cec..dd1ca2648 100644
--- a/addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/ExtensionParamMiner.java
+++ b/addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/ExtensionParamMiner.java
@@ -131,6 +131,7 @@ public class ExtensionParamMiner extends ExtensionAdaptor {
                     e -> {
                         showParamMinerDialog(null);
                     });
+            menu.setIcon(getIcon());
         }
         return menu;
     }
diff --git a/addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/gui/ParamMinerDialog.java b/addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/gui/ParamMinerDialog.java
index 9c3b17561..0c83c8a79 100644
--- a/addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/gui/ParamMinerDialog.java
+++ b/addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/gui/ParamMinerDialog.java
@@ -324,6 +324,7 @@ public class ParamMinerDialog extends StandardFieldsDialog {
         JList<Methods> list = new JList<>(methods.toArray(new Methods[0]));
         list.setLayoutOrientation(JList.HORIZONTAL_WRAP);
         list.setVisibleRowCount(1);
+        list.setToolTipText(Constant.messages.getString("paramminer.dialog.urlguess.methods.tooltip"));
         JPanel panel = createCustomPanel(label, list);
         panel.setVisible(isVisible);
         this.addCustomComponent(tabIndex, panel);
diff --git a/addOns/paramminer/src/main/resources/org/zaproxy/addon/paramminer/resources/Messages.properties b/addOns/paramminer/src/main/resources/org/zaproxy/addon/paramminer/resources/Messages.properties
index 7290ca1eb..c9df37a4a 100644
--- a/addOns/paramminer/src/main/resources/org/zaproxy/addon/paramminer/resources/Messages.properties
+++ b/addOns/paramminer/src/main/resources/org/zaproxy/addon/paramminer/resources/Messages.properties
@@ -60,6 +60,7 @@ paramminer.dialog.urlguess.methods.get = GET
 paramminer.dialog.urlguess.methods.post = POST
 paramminer.dialog.urlguess.methods.xml = XML
 paramminer.dialog.urlguess.methods.json = JSON
+paramminer.dialog.urlguess.methods.tooltip = Multi-select with CTRL + Click (or CMD + Click).

 paramminer.dialog.skip.boring.headers = Skip Boring Headers:
 paramminer.dialog.threadpool.size = Number of Threads:

```

**Comment by ArkaprabhaChakraborty** [comment]
I think this can be merged :)

**Comment by thc202** [comment]
This needs to be rebased.

**Comment by ArkaprabhaChakraborty** [comment]
Are there anymore related necessary changes for this PR tho? I'll rebase it tomorrow morning :) but before that I would like to make the changes necessary :).

**Comment by ArkaprabhaChakraborty** [comment]
PR title also needs to be changed I guess.

File: addOns/paramminer/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ All notable changes to this add-on will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
+Rename multiple options in GUI and documentation.
```
Comment (thc202) [inline addOns/paramminer/CHANGELOG.md:7]: Once rebased this should be under a `### Changed` section, should also start with a hyphen.

File: addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/gui/ParamMinerDialog.java [hunk 2]
```diff
@@ -309,12 +309,12 @@ public void init(HttpMessage target) {
 
     @Override
     public String getHelpIndex() {
-        return "paramminer.dialog";
+        return "paramdigger.dialog";
```
Comment (thc202) [inline addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/gui/ParamMinerDialog.java:312]: This was not yet changed in the map.

File: addOns/paramminer/src/main/resources/org/zaproxy/addon/paramminer/resources/Messages.properties [hunk 3]
```diff
@@ -1,81 +1,82 @@
...
-paramminer.desc = Parameter Miner
-paramminer.error.nofile = File not found : {0}
-paramminer.panel.msg = <html>Param Miner panel</html>
-paramminer.panel.title = Param Miner
-paramminer.popup.msg = Example right click menu invoked for:\n {0}
-paramminer.popup.title = Param Miner...
-paramminer.topmenu.tools.msg = Param Miner: Useful for finding cache poisoning vulnerabilities
-paramminer.topmenu.tools.title = Param Miner
+paramdigger.api.action.helloWorld = Logs "hello world called" with debug level.
+paramdigger.desc = Parameter Miner
```
Comment (thc202) [inline addOns/paramminer/src/main/resources/org/zaproxy/addon/paramminer/resources/Messages.properties:2]: Needs to be updated.

File: addOns/paramminer/src/main/resources/org/zaproxy/addon/paramminer/resources/Messages.properties [hunk 4]
```diff
@@ -1,81 +1,82 @@
...
+paramdigger.error.nofile = File not found : {0}
+paramdigger.panel.msg = <html>Param Digger panel</html>
+paramdigger.panel.title = Param Digger
+paramdigger.popup.msg = Example right click menu invoked for:\n {0}
```
Comment (thc202) [inline addOns/paramminer/src/main/resources/org/zaproxy/addon/paramminer/resources/Messages.properties:6]: Is this used? Maybe delete now.

**Review by kingthorin — APPROVED** [review]
Tks

File: addOns/paramminer/src/main/javahelp/org/zaproxy/addon/paramminer/resources/help/contents/paramdigger.html [hunk 5]
```diff
@@ -2,13 +2,13 @@
 <HTML>
 <HEAD>
 <TITLE>
-Parameter Miner
+Parameter Discovery Add On
```
Comment (thc202) [inline addOns/paramminer/src/main/javahelp/org/zaproxy/addon/paramminer/resources/help/contents/paramdigger.html:5]: I'd suggest keeping the same name everywhere, `Parameter Digger`.

File: addOns/paramminer/src/main/resources/org/zaproxy/addon/paramminer/resources/Messages.properties [hunk 6]
```diff
@@ -1,81 +1,81 @@
...
-paramminer.desc = Parameter Miner
-paramminer.error.nofile = File not found : {0}
-paramminer.panel.msg = <html>Param Miner panel</html>
-paramminer.panel.title = Param Miner
-paramminer.popup.msg = Example right click menu invoked for:\n {0}
-paramminer.popup.title = Param Miner...
-paramminer.topmenu.tools.msg = Param Miner: Useful for finding cache poisoning vulnerabilities
-paramminer.topmenu.tools.title = Param Miner
+paramdigger.api.action.helloWorld = Logs "hello world called" with debug level.
+paramdigger.desc = Parameter Discovery Tool
```
Comment (thc202) [inline addOns/paramminer/src/main/resources/org/zaproxy/addon/paramminer/resources/Messages.properties:2]: Better keep the same name.

File: addOns/paramminer/src/main/javahelp/org/zaproxy/addon/paramminer/resources/help/toc.xml [hunk 7]
```diff
@@ -6,10 +6,10 @@
 <toc version="2.0">
 	<tocitem text="ZAP User Guide" tocid="toplevelitem">
 		<tocitem text="Add Ons" tocid="addons">
-			<tocitem text="Parameter Miner" image = "param-miner-icon" target="paramminer">
-				<tocitem text="About" target="paramminer.about"/>
-				<tocitem text="Dialog" target="paramminer.dialog"/>
-				<tocitem text="Tab" target="paramminer.tab"/>
+			<tocitem text="Parameter Miner" image = "param-miner-icon" target="paramdigger">
```
Comment (thc202) [inline addOns/paramminer/src/main/javahelp/org/zaproxy/addon/paramminer/resources/help/toc.xml:9]: Text needs to be updated.

File: addOns/paramminer/src/main/javahelp/org/zaproxy/addon/paramminer/resources/help/map.jhm [hunk 8]
```diff
@@ -5,8 +5,8 @@
 
 <map version="1.0">
 	<mapID target="param-miner-icon" url="contents/images/hard-hat-mine.png" />
```
Comment (thc202) [inline addOns/paramminer/src/main/javahelp/org/zaproxy/addon/paramminer/resources/help/map.jhm:7]: Worth updating the target as well.

File: addOns/paramminer/src/main/javahelp/org/zaproxy/addon/paramminer/resources/help/index.xml [hunk 9]
```diff
@@ -4,7 +4,7 @@
          "http://java.sun.com/products/javahelp/index_2_0.dtd">
 
 <index version="2.0">
-	<indexitem text="Param Miner" target="paramminer" />
-  <indexitem text="Param Miner dialog" target="paramminer.dialog" />
-  <indexitem text="Param Miner tab" target="paramminer.tab" />
+	<indexitem text="Param Miner" target="paramdigger" />
+  <indexitem text="Param Miner dialog" target="paramdigger.dialog" />
+  <indexitem text="Param Miner tab" target="paramdigger.tab" />
```
Comment (thc202) [inline addOns/paramminer/src/main/javahelp/org/zaproxy/addon/paramminer/resources/help/index.xml:9]: The text needs update.

File: addOns/paramminer/src/main/javahelp/org/zaproxy/addon/paramminer/resources/help/contents/tab.html [hunk 10]
```diff
@@ -44,20 +44,20 @@ <h1>Param Miner tab</h1>
...
+	<p>For each Param Digger message, shown under the Output tab, you can see details of the Parameter found and the 
         reason why it was detected.</p>
 
 	<h2>See also</h2>
 	<table>
 		<tr>
 			<td>&nbsp;&nbsp;&nbsp;&nbsp;</td>
-			<td><a href="paramminer.html">Param Miner</a></td>
-			<td>for an overview of the Param Miner.</td>
+			<td><a href="paramminer.html">Param Digger</a></td>
```
Comment (thc202) [inline addOns/paramminer/src/main/javahelp/org/zaproxy/addon/paramminer/resources/help/contents/tab.html:54]: Link needs to be updated.

File: addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/ExtensionParamMiner.java [hunk 11]
```diff
@@ -34,7 +34,7 @@
 public class ExtensionParamMiner extends ExtensionAdaptor {
 
     public static final String NAME = "ExtensionParamMiner";
```
Comment (kingthorin) [inline addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/ExtensionParamMiner.java:36]: Here too

Comment (thc202) [reply addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/ExtensionParamMiner.java:36]: I think the code/classes will be renamed in another PR.

**Comment by ArkaprabhaChakraborty** [comment]
that should be it I guess :)

**Comment by kingthorin** [comment]
I’d suggest doing a search for liner within the whole directory.

**Comment by ArkaprabhaChakraborty** [comment]
> I’d suggest doing a search for miner within the whole directory.

there are many areas ... possibly no more in GUI.. or help docs

**Comment by ArkaprabhaChakraborty** [comment]
I'll be making more PRs to change things possibly they would be caught in next PRs

File: addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/ExtensionParamMiner.java (hunk 11 above)
Comment (thc202) [reply addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/ExtensionParamMiner.java]: If you are changing the name change the class as well.

Comment (thc202) [reply addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/ExtensionParamMiner.java]: Anyway.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

**Pull request merged**